sys/include/net/ng_fib/ng_universal_address.h

```diff
@@ -0,0 +1,126 @@
+
+
+/**
+ * @brief The container descriptor used to identify a universal address entry
+ */
+typedef struct universal_address_container_t {
+    uint8_t use_count;                       /**< The number of entries link here */
+    uint8_t address_size;                    /**< Size in bytes of the used genereic address */
+    uint8_t address[UNIVERSAL_ADDRESS_SIZE]; /**< the genereic address data */
+} universal_address_container_t;
+
+/**
+ * @brief initializes the datastructure for the entries
+ */
+void universal_address_init(void);
+
+/**
+ * @brief resets the usecoumt for all entries
+ */
+void universal_address_reset(void);
+
+/**
+ * @brief Adds a given address to the universal address entries
+ *        if the entry already exists, the use_count will be increased.
+ *
+ * @param[in] addr       pointer to the address
+ * @param[in] addr_size  the number of bytes required for the address entry
+ *
+ * @return pointer to the universal_address_container_t containing the address on success
+ *         NULL if the address could not be inserted
+ */
+universal_address_container_t *universal_address_add(uint8_t *addr, size_t addr_size);
+
+/**
+ * @brief Adds a given container from the universal address entries
+ *        if the entry exists, the use_count will be decreased.
+ *
+ * @param[in] entry  pointer to the universal_address_container_t to be removed
+ */
+void universal_address_rem(universal_address_container_t *entry);
+
+/**
+ * @brief copies the address from the given container to the provided pointer
+ *
+ * @param[in] entry          pointer to the universal_address_container_t
+ * @param[out] addr          pointer to store the address entry
+ * @param[in, out] addr_size pointer providing the size of available memory on addr
+ *                           this value is overwritten with the actual size required
+ *
+ * @return UNIVERSAL_ADDRESS_SUCCESS if the address is copied to the addr destination
+ *         UNIVERSAL_ADDRESS_ERROR_MEMORY if the size is unsufficient for copy
+ */
+int universal_address_get_address(universal_address_container_t *entry,
+                                  uint8_t *addr, size_t *addr_size);
+
+/**
+ * @brief determines if the entry equals the provided address
+ *
+ * @param[in] entry       pointer to the universal_address_container_t for compare
+ * @param[in] addr        pointer to the address for compare
+ * @param[in, out] addr_size   the number of bytes used for the address entry
+ *                             on sucessfull return this value is overwritten
+ *                             with the number of matching bytes till the
+ *                             first of trailing `0`s indicating a prefix
+ *
+ * @return UNIVERSAL_ADDRESS_SUCCESS if the entries are equal or comperable
+ *         UNIVERSAL_ADDRESS_ERROR_MEMORY if the given address sizes diverges
+ *         UNIVERSAL_ADDRESS_NO_HIT if the given adresses do not match
```

Comment:
Can't this be as one normally would expect for e.g. `memcmp()`/`strcmp()`: 0 on hit, non-zero on difference.